account_move_locking/models/date_range.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# © 2016 Camptocamp SA (Matthieu Dietrich)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from openerp import _, api, exceptions, fields, models
+
+
+class DateRange(models.Model):
+    _inherit = "date.range"
+
+    @api.depends('locked_journal_ids')
+    def _get_lock_state(self):
+        journal_obj = self.env['account.journal']
+        for range in self:
+            all_journals = journal_obj.search(
+                [('company_id', '=', range.company_id.id)])
+            if not range.locked_journal_ids:
+                range.lock_state = 'unlocked'
+            elif range.locked_journal_ids == all_journals:
+                range.lock_state = 'locked'
+            else:
+                range.lock_state = 'partial'
+
+    lock_state = fields.Selection(
+        [('unlocked', 'Unlocked'),
+         ('partial', 'Partially locked'),
+         ('locked', 'Locked')],
+        compute="_get_lock_state",
+        readonly=True,
+        store=True)
+    locked_journal_ids = fields.Many2many(
+        'account.journal',
+        rel='account_date_range_lock_journal',
+        string="Locked journals")
+
+    @api.model
+    def create(self, vals):
+        # Check if fiscal year is locked; if so, add all journals
+        # to set is as locked
```

Comment:
I do not feel comfortable with this behavior since you apply a locking strategy proper to the accounting area to all `date.range`. Therefore, it's no more possible to delete/modify any kind of date.range if these are included in a FY period.... IMO we can't enforce this rule for all kind of `date.range`.